### PR TITLE
Replace microbundle with rollup

### PR DESCRIPTION
**Pull request by binoy14**
I don't believe this is a breaking changes we did not change the file but it's hard to tell :D so just to be save maybe bundle this in v7 release? 

- Replace microbundle with Rollup
- Added size-limit to cover all bundle types
- Added a github actions to report size changes in PRs

### CODE REVIEW

**Comment by hartzis** [comment]
- [ ] ❓ should we remove the `"esmodule": "./dist/react-swipeable.modern.js",` output in `package.json`?

**Comment by binoy14** [comment]
> * should we remove the `"esmodule": "./dist/react-swipeable.modern.js",` output in `package`

I believe we should still keep it since most modern frameworks are moving to ESmodule, example NextJS so the bundlers have an option to pick whatever the app is using. Not an expert here but I think that's how this works :D

**Comment by hartzis** [comment]
Reference for exports when reviewing and deciding
- https://github.com/developit/microbundle/issues/921#issuecomment-1003104604

**Comment by rschristian** [comment]
Yes, `"esmodule"` should be removed as a key. It is the precursor to the modern `"exports"` but is very much legacy at this point, next to nothing actually supports consuming it. Microbundle was pushing it before `"exports"` was solidified and picked up by Node. We've since removed it from the docs.

> I believe we should still keep it since most modern frameworks are moving to ESmodule,

Supporting ESM is very different from reading the `"esmodule"` key.

**Comment by binoy14** [comment]
> I believe we should still keep it since most modern frameworks are moving to ESmodule, example NextJS so the bundlers have an option to pick whatever the app is using. Not an expert here but I think that's how this works :D

I misread the comment earlier, I updated it by removing `esmodule` from package.json thanks!

File: package.json [hunk 1]
```diff
@@ -86,16 +96,15 @@
...
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
     "rimraf": "^3.0.2",
-    "rollup": "^1.1.2",
-    "rollup-plugin-babel": "^4.3.2",
+    "rollup": "^2.60.0",
     "size-limit": "^4.6.0",
     "ts-jest": "^26.4.4",
     "ts-loader": "^8.0.1",
+    "tslib": "^2.3.1",
```
Comment (hartzis) [inline package.json:107]: [ ] do we need this dep?

File: .github/workflows/size.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: size
+on: [pull_request]
+jobs:
+  size:
+    runs-on: ubuntu-latest
+    env:
+      CI_JOB_NUMBER: 1
+    steps:
+      - uses: actions/checkout@v1
+      - uses: andresz1/size-limit-action@v1
```
Comment (hartzis) [inline .github/workflows/size.yml:10]: 🎉 this is awesome

File: rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      format: "umd",
+      sourcemap: true,
+      globals: {
+        react: 'React'
+      }
+    },
+    external: ["react"],
+    plugins: [
+      typescript(),
+      commonjs()
```
[the remark this answers is not in the document]
Comment (binoy14) [reply rollup.config.js:21]: That's for backwards compatibility with commonjs, we can make a decision to always be es modules but that will probably not enable few users who are not 100% es modules to use this package. The current microbundle produces both so this keeps it in sync with it.

**Review by hartzis — APPROVED** [review]

**Pull request merged**